### PR TITLE
Refactor 'create run' containers to functional components

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

#2000 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/containers/CreateTaskRun/CreateTaskRun.js [hunk 1]
```diff
@@ -223,29 +257,31 @@ class CreateTaskRun extends React.Component {
...
-  handleClose = () => {
-    const { defaultNamespace: namespace, history } = this.props;
-    const { kind, taskName } = this.getTaskDetails();
+  function handleClose() {
+    const { kind: taskKind, taskName } = getTaskDetails();
     let url = urls.taskRuns.all();
-    if (taskName && namespace !== ALL_NAMESPACES) {
-      url = urls.taskRuns[kind === 'ClusterTask' ? 'byClusterTask' : 'byTask']({
-        namespace,
+    if (taskName && defaultNamespace !== ALL_NAMESPACES) {
```
Comment (briangleeson) [inline src/containers/CreateTaskRun/CreateTaskRun.js:269]: I notice here that we are comparing `defaultNamespace` from props in the if statement, whereas in [CreatePipelineRun in the handleClose function](https://github.com/tektoncd/dashboard/pull/2037/files?diff=split&w=1#diff-a76903232a0bbce3293cb530f3aac343e2b5d6dda21048219bb2467ce9b5027dR255) we check against `namespace` from the state.

Should these match each other? The old code suggests we should be checking against `defaultNamespace` in both cases?

**Review by briangleeson — CHANGES_REQUESTED** [review]
One query about handleClose functions

File: src/containers/CreateTaskRun/CreateTaskRun.js (hunk 1 above)
Comment (AlanGreene) [reply src/containers/CreateTaskRun/CreateTaskRun.js:269]: nice catch, thanks. Fixed

**Pull request merged**